### ISSUE TITLE
If moving the same event to a later date, make sure alarms fire again even if they already fired once

### ISSUE THREAD

**Issue by brong** [cyrusimap/cyrus-imapd#1979]
The user reporting this has an event that recurs monthly on the 14th of the month at 10am, starting in March.  User moved the May occurrence that event to the 15th and got a reminder as expected, then moved the same event to the 16th of May.  On the 16th, the reminder didn't fire.

I haven't investigated exactly what's happening in the Cyrus code, but I suspect that we had annotated that the particular recurrence had already had its alarm sent.

**Comment by rsto** [comment]
I've tried reproducing this issue but can't. A Cassandane test similar to the scenario in this issue's description runs OK: https://github.com/cyrusimap/cassandane/commit/b97ba9cbe78e48d10ee0cbdb1f43a73b907e3460

I only could identify edge cases around shifting events within windows of a few seconds, namely the 10-second default calalarmd window.

In addition, I ran tests in FastMail production (using the regular UI), all showing the same, sane result. I'll run an additional multi-day test for the next 2 days and will close this issue if it passes as well.

**Comment by brong** [comment]
Right, OK. I'll see if I can find a user who's failing to move due to
alarms that didn't run properly and dig out the details of that event!

On Tue, 23 May 2017, at 23:03, Robert Stepanek wrote:
> I've tried reproducing this issue but can't. A Cassandane test similar
> to the scenario in this issue's description runs OK:
> cyrusimap/cassandane@b97ba9c> I only could identify edge cases around shifting events within windows
> of a few seconds, namely the 10-second default calalarmd window.> In addition, I ran tests in FastMail production (using the regular
> UI), all showing the same, sane result. I'll run an additional multi-
> day test for the next 2 days and will close this issue if it passes
> as well.> — You are receiving this because you authored the thread. Reply to
> this email directly, view it on GitHub[1], or mute the thread[2].> 

--
  Bron Gondwana
  brong@fastmail.fm



Links:

  1. https://github.com/cyrusimap/cyrus-imapd/issues/1979#issuecomment-303390921

**Comment by rsto** [comment]
This issue is still on. I recently had to disable its previously working test: https://github.com/cyrusimap/cassandane/commit/5d9587cee72b4b1b81d6181d56ef6811440eef9d

haven't identified the root cause, yet.

**Comment by rsto** [comment]
Could this be related to https://github.com/cyrusimap/cyrus-imapd/issues/2211 ?

**Comment by rsto** [comment]
@ksmurchison Bron and I are sifting through the backlog. Do you have an idea on this one?

**Comment by ksmurchison** [comment]
@rsto  No clue.  So, your Cass test used to work and now it fails?

**Comment by rsto** [comment]
@ksmurchison No, I don't have a test for this. We just thought you might have a clue if that issue is still a thing.

**Comment by ksmurchison** [comment]
Do we know if this is an issue for ALL calendars or just shared ones?

**Comment by rjbs** [comment]
:robot: I'm closing this issue by automation.

This issue has been sitting untouched for over six years.  We're trying to get a good handle on our open issues and pull requests, so that we can reliably use them as a backlog of things we hope to work on.  If you are confident that this issue should be reconsidered, reopen it and explain why.

Remember, though, that there are only so many developers working on Cyrus, and we'd like to focus on the most important and/or low-hanging work.

Before you reopen this issue, please check that this issue is still present in the latest release of Cyrus, or the main development branch. An issue is considerably more likely to get resolved if you can provide a pull request which reproduces the issue in a Cassandane test.
